addons/project/static/src/actions/client_actions.js

```diff
@@ -25,34 +29,94 @@ export function displayTemplateNotificationAction(env, action) {
             },
         ],
     });
-    return params.next;
 }
 
-registry
-    .category("actions")
-    .add("project_show_template_notification", displayTemplateNotificationAction);
+export function showUndoConfirmationDialog(
+    env,
+    { model, recordId, bodyMessage, confirmLabel, undoMethod = "action_undo_convert_to_template" }
+) {
+    env.services.dialog.add(ConfirmationDialog, {
+        body: _t(bodyMessage),
+        confirmLabel: _t(confirmLabel),
+        confirm: async () => {
+            const action = await env.services.orm.call(model, undoMethod, [recordId]);
+            await env.services.action.doAction(action);
+        },
+        cancelLabel: _t("Discard"),
+        cancel: () => {},
+    });
+}
 
-export function displayTemplateUndoConfirmationDialog(env, action) {
-    const params = action.params || {};
-    const taskId = params.task_id;
+export function showConfirmationDialog(env,
+    { model, recordId, bodyMessage, confirmLabel, method = "convert_project_to_template" }
+) {
     env.services.dialog.add(ConfirmationDialog, {
-        body: _t(
-            "This task is already a template. Do you want to convert it back to a regular task?"
-        ),
-        confirmLabel: _t("Convert to Task"),
+        body: _t(bodyMessage),
+        confirmLabel: _t(confirmLabel),
```

Comment:
It will not work, we should explicitly give the string to the translation function.

Reply:
Actually, we could remove `_t` here seems the content is already translated, no?